code/clue.js

```diff
@@ -1,59 +1,304 @@
+};
+
+//----- Room card -----//
+const pickRoomLoader = () => {
+  document.getElementById("loader3").classList.add("animated-loader");
+  setTimeout(showPickRoom, 2000);
+};
+
+document.getElementById("roomCard").addEventListener("click", pickRoomLoader); 
+
+const showPickRoom = () => {
+  mystery.room = randomSelector(rooms);
+  document.getElementById("loader3").style.display = "none"
+  document.getElementById('roomCard').removeEventListener('click', pickRoomLoader);
+
+  document.getElementById("roomCard").style.background = mystery.room.color;
+  document.getElementById("roomName").innerHTML = `${mystery.room.name}`;
+  document.getElementById("roomImage").src = mystery.room.image;
+  document.getElementById("roomLastSeen").innerHTML = `Last seen there: ${mystery.room.lastseen}`;
+  document.getElementById("roomfave").innerHTML = `Favourite room: ${mystery.room.favespot}`;
+
+  document.getElementById("cardDetails3").style.display = "flex";
+};
+
+//----- Function for on button click reveal mystery or get a message asking you to choose each card to then beable to reveal the mystery-----//
+function revealMystery() {
+  if (mystery.killer.hasOwnProperty("name") === true) {
+    document.getElementById("mystery").innerHTML = `The murder was commited by ${mystery.killer.name} with the ${mystery.weapon.name.toLowerCase()} in the ${mystery.room.name.toLowerCase()}`;
+    document.getElementById("revealMysteryButton").innerHTML = "PLAY AGAIN!";
+    document.getElementById("revealMysteryButton").onclick = resetPage;
+    document.getElementById("choose-card-prompt").style.display = "none";
+
+  } else {
+    document.getElementById("choose-card-prompt").innerHTML = "Please select one of each card to reveal the mystery!";
+  }
+};
 
-// This function will be invoked when you click on the killer card.
-const pickKiller = () => {
-  // This will randomly select a killer from the suspects. And add that to the mystery object.
-  mystery.killer = randomSelector(suspects)
+//----- Shuffle function -----//
+// Allows for the weapons to be shuffled between the suspects
+const shuffleFavouriteWeapon = () => {
+  let copyWeapons = weapons.slice(0); //Creates shallow copy of weapons array
+  for (let i = 0; i < suspects.length; i++ ) {
+    let randomIndex = Math.floor(Math.random() * copyWeapons.length);
+    let randomWeapon = copyWeapons.splice(randomIndex, 1);
+    suspects[i].favouriteWeapon = randomWeapon[0];
+  }
+};
 
-  // This will change the background color of the card to the one connected to the chosen killer and show the full name of the killer. Feel free to add more things to show about the killer.
-  document.getElementById('killerCard').style.background = mystery.killer.color
-  document.getElementById(
-    'killerName'
-  ).innerHTML = `${mystery.killer.firstName} ${mystery.killer.lastName}`
-}
+//Function for re-loading the page
+const resetPage = () => {
+  location.reload();
```

Comment:
Hmm, wouldn't it be better to just clear mystery object and selected cards?

Reply:
@EmOlsz How would I go about doing this? Sorry I'm uncertain of how to do this.